expo/expo
# [config-plugins][updates] Add support for config.updates.useClassicUpdates defaulting behavior

# Why

Pulls in the new `updates.useClassicUpdates` property from https://github.com/expo/universe/pull/12130.

Part of https://linear.app/expo/issue/ENG-6812/make-eas-update-default-in-sdk-49.

This flag will be used to:
- instruct config plugin to default to classic updates (this PR)
- override default development manifest type to classic (being changed in https://github.com/expo/expo/pull/22168)

This PR is a breaking change for the next SDK version. This requires the developer to add this flag if they currently don't use EAS update and wish to keep using classic updates.

# How

Pull in schema.
Make changes to updates config plugin and tests.

# Test Plan

Run tests.

Manual test:
1. `et gba -n testwat @expo/config-plugins @expo/config-types`
2. `eas build:configure`
3. `eas build --profile development`
4. Inspect build, ensure that no update URL is set in AndroidManifest.xml and enabled is false
5. Add updates.useClassicUpdates in app.json
6. `npx expo prebuild`, see AndroidManifest.xml has classic updates url

# Checklist
